src/hasher512.cpp

```diff
@@ -160,9 +160,9 @@ namespace libtorrent
 #endif
 		}
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), data.size());
+		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), std::uint32_t(data.size()));
```

Comment:
ok, i will revert std::uint32_t cast from this pull request and will make separate pull request which will make internal hash declaration match libcrypt api.